src/core/lwt.mli

```diff
@@ -796,6 +796,25 @@ val try_bind : (unit -> 'a t) -> ('a -> 'b t) -> (exn -> 'b t) -> 'b t
       performing any operation on one is equivalent to performing it on the
       other. *)
 
+val dont_wait : (unit -> unit t) -> (exn -> unit) -> unit
+(** [Lwt.dont_wait f handler] applies [f ()], which returns a promise, and then
+    makes it so that if the promise is {{: #TYPEt} {e rejected}}, the exception
+    is passed to [handler].
+
+    In addition, if [f ()] raises an exception, it is also passed to [handler].
+
+    As the name implies, [dont_wait (fun () -> <e>) handler] is a way to
+    evaluate the expression [<e>] (which typically has asynchronous
+    side-effects) {e without waiting} for the resolution of the promise [<e>]
+    evaluates to.
+
+    [dont_wait] is meant as an alternative to {!async} with a local, explicit,
+    predictable exception handler.
+
+    Note that [dont_wait f h] causes [f ()] to be evaluated immediately.
+    Consequently, the non-yielding/non-pausing prefix of the body of [f] is
+    evaluated immediately. *)
+
```

Comment:
Would it be nice to modify documentation of `async` and/ore `ignore_result` to propose `dont_wait` as a safer alternative to use?

Reply:
I added a mention of `dont_wait` in the documentation of `async`.

Because `async` and `ignore_result` are very similar (and thus it makes sense to point `ignore_result` specifically to `async`) I haven't added the mention in the documentation of `ignore_result`.

I also don't necessarily want to say that `dont_wait` is "safer" than `async`. There are very valid uses of `async`. (We departed from it in the Tezos code-base because the code-base and development team are large and we are trying to go for more explicit error/exception management. But for smaller projects it can make sense to use `async`.)

Reply:
I am fine with that, thanks!